features/step_definitions/collection.js

```diff
@@ -0,0 +1,273 @@
+            });
+    });
+
+    Given(/^they have a collection "anotherCollection"$/, function step() {
+        return createCollection.call(this, { name: 'anotherCollection' })
+            .then((response) => {
+                Assert.oneOf(response.statusCode, [409, 201]);
+
+                if (response.statusCode === 201) {
+                    this.secondCollectionId = response.body.id;
+                } else {
+                    const str = response.body.message;
+
+                    [, this.secondCollectionId] = str.split(': ');
+                }
+            });
+    });
+
+    When(/^they fetch all their collections$/, function step() {
+        return request({
+            uri: `${this.instance}/${this.namespace}/collections`,
+            method: 'GET',
+            auth: {
+                bearer: this.jwt
+            },
+            json: true
+        }).then((response) => {
+            Assert.strictEqual(response.statusCode, 200);
+            this.collections = response.body;
+        });
+    });
+
+    Then(/^they can see those collections$/, function step() {
+        Assert.strictEqual(this.collections.length, 2);
+        Assert.strictEqual(this.collections[0].name, 'anotherCollection');
+        Assert.strictEqual(this.collections[1].name, 'myCollection');
+    });
+
+    When(/^they delete that collection$/, function step() {
+        return this.ensurePipelineExists({ repoName: this.repoName })
+            .then(() =>
+                request({
+                    uri: `${this.instance}/${this.namespace}/collections/${this.firstCollectionId}`,
+                    method: 'DELETE',
+                    auth: {
+                        bearer: this.jwt
+                    },
+                    json: true
+                })
+            )
+            .then((response) => {
+                Assert.strictEqual(response.statusCode, 204);
+            });
+    });
+
+    Then(/^that collection no longer exists$/, function step() {
+        return request({
+            uri: `${this.instance}/${this.namespace}/collections/${this.firstCollectionId}`,
+            method: 'GET',
+            auth: {
+                bearer: this.jwt
+            },
+            json: true
+        }).then((response) => {
+            Assert.strictEqual(response.statusCode, 404);
+            this.firstCollectionId = null;
+        });
+    });
+
+    Then(/^that pipeline still exists$/, function step() {
+        return request({
+            uri: `${this.instance}/${this.namespace}/pipelines/${this.pipelineId}`,
+            method: 'GET',
+            auth: {
+                bearer: this.jwt
+            },
+            json: true
+        }).then((response) => {
+            Assert.strictEqual(response.statusCode, 200);
+            Assert.strictEqual(response.body.scmRepo.name, `${this.testOrg}/${this.repoName}`);
+        });
+    });
+
+    When(/^they create another collection "myCollection"$/, function step() {
```

Comment:
Maybe we should rename this scenario to "When they create another collection with the same name 'myCollection'"